pkg/certificate/providers/config.go

```diff
@@ -237,19 +237,21 @@ func (c *Config) getTresorOSMCertificateManager() (certificate.Manager, debugger
 		return nil, nil, errors.Errorf("Failed to synchronize certificate on Secrets API : %v", err)
 	}
 
-	certManager, err := tresor.NewCertManager(
+	tresorClient, err := tresor.New(
 		rootCert,
 		rootCertOrganization,
-		c.cfg,
-		c.cfg.GetServiceCertValidityPeriod(),
 		c.cfg.GetCertKeyBitSize(),
-		c.msgBroker,
 	)
 	if err != nil {
 		return nil, nil, errors.Errorf("Failed to instantiate Tresor as a Certificate Manager")
 	}
 
-	return certManager, certManager, nil
+	tresorCertManager, err := certificate.NewManager(rootCert, tresorClient, c.cfg.GetServiceCertValidityPeriod(), c.msgBroker)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error instantiating osm certificate.Manager for Vault cert-manager : %w", err)
+	}
+
```

Comment:
remember that you'll need to call:

	`rotor.New(tresorCertManager).Start(checkCertificateExpirationInterval)`